src/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Text } from '../typography/typography';
+
+const Checkbox = (props) => {
+  const [checked, setChecked] = useState(true);
+
+  const handleChange = () => {
+    setChecked(!checked);
+    props.onChange(checked);
+  };
+
+  const HiddenCheckbox = styled.input.attrs({
+    type: 'checkbox',
+    onChange: handleChange,
+  })`
+    height: 1px;
+    width: 1px;
+    margin: -1px;
+    overflow: hidden;
+    position: absolute;
+    visibility: hidden;
+  `;
+
+  const Icon = styled.svg`
+    display: inline-block;
+    fill: none;
+    margin-top: 5px;
+    margin-left: 5px;
+  `;
+
+  const StyledCheckbox = styled.div`
+    display: inline-block;
+    width: 24px;
+    height: 24px;
+
+    background: #f7f8fa;
+    border-radius: 4px;
+
+    ${Icon} {
+      visibility: ${checked ? 'visible' : 'hidden'};
+    }
+  `;
+
+  const CheckboxContainer = styled.div`
+    display: flex;
+    align-items: center;
+  `;
+
+  const HiddenCheckboxLabel = styled.label`
+    display: inline-block;
+  `;
+
+  const LabelText = styled(Text)`
```

Comment:
The `Text` component has a `margin-bottom` of 16 pixels which does not quite fit here. I would also not recommend to nest a `p` element (`<p>` is a block element, `Text` is `styled.p`) inside a `label` element (`<span>` is an inline element). We should almost never nest block elements inside inline elements. Here a brief introduction into the "block"/"inline" topic: https://www.w3schools.com/html/html_blocks.asp

I suggest to create a new typography type for this special case that does not have any outer spacing as well as uses an inline element (`<span>` in this case) as its base.